storage/filestore.go

```diff
@@ -221,7 +221,7 @@ func (f *FilesystemStore) Set(name string, meta []byte) error {
 	os.RemoveAll(fp)
 
 	// Write the file to disk
-	if err = ioutil.WriteFile(fp, meta, f.perms); err != nil {
+	if err = ioutil.WriteFile(fp, meta, f.perms&notary.PermExecutableMask); err != nil {
```

Comment:
I'm wondering if instead of having to perform this bitwise and `&` at every possible write point, that if we instead create a new unexported field for `FilesystemStore` for file permissions without the executable bits set.  

For example, this write could become `ioutil.WriteFile(fp, meta, f.nonExecutePerms)`